be/src/storage/lake/tablet.cpp

```diff
@@ -191,8 +191,9 @@ int64_t Tablet::data_size() {
 }
 
 size_t Tablet::num_rows() const {
-    int64_t version = _version_hint;
-    auto num_rows = _mgr->get_tablet_num_rows(_id, &version);
+    // set_version_hint should be called before to avoid list tablet metadata
```

Comment:
How can this be assured?

Reply:
It can only be guaranteed by the caller, otherwise `get_tablet_num_rows` need to list tablet metadata